### PR TITLE
(DataTable) Support columns with arbitrary content #56

### PR DESCRIPTION
**Pull request by vyacheslav-pushkin**

### CODE REVIEW

File: packages/cuba-react/src/ui/table/DataTable.tsx [hunk 1]
```diff
@@ -49,12 +52,27 @@ export interface DataTableProps<E> extends MainStoreInjected {
...
+export interface ColumnDefinition<E> {
+  field?: string,
+  columnProps: ColumnProps<E>
 }
 
 @injectMainStore
 @observer
 export class DataTable<E> extends React.Component<DataTableProps<E>> {
+
+  static readonly NO_COLUMN_DEF_ERROR = 'You need to provide either `columnDefinitions` or `fields` prop to the `DataTable`';
```
Comment (vadimbasko) [inline packages/cuba-react/src/ui/table/DataTable.tsx]: When we throw this error, is it possible to add info about context for ease of debug. E.g. entity name or may be something else about current DataTable?

File: packages/cuba-react/src/ui/table/DataTable.tsx [hunk 2]
```diff
@@ -353,28 +388,53 @@ export class DataTable<E> extends React.Component<DataTableProps<E>> {
...
+          operator: this.operatorsByProperty.get(propertyName),
+          onOperatorChange: this.handleFilterOperatorChange,
+          value: this.valuesByProperty.get(propertyName),
+          onValueChange: this.handleFilterValueChange,
+          enableSorter: true,
+          mainStore: this.props.mainStore!,
+          customFilterRef: (instance: any) => this.customFilters.set(propertyName, instance)
+        });
+
+        const customizedColumn = {
```
Comment (vadimbasko) [inline packages/cuba-react/src/ui/table/DataTable.tsx]: IDEA highlights warning as - Local variable 'customizedColumn' is redundant

File: packages/cuba-react/src/ui/table/DataTable.tsx (hunk 1 above)
Comment (vyacheslav-pushkin) [reply packages/cuba-react/src/ui/table/DataTable.tsx]: done

File: packages/cuba-react/src/ui/table/DataTableCustomFilter.tsx [hunk 3]
```diff
@@ -105,6 +105,10 @@ class DataTableCustomFilterComponent<E extends WithId>
     }
   }
 
+  @computed get errorContext(): string {
```
Comment (web-devel) [inline packages/cuba-react/src/ui/table/DataTableCustomFilter.tsx]: Should it be computed?

**Pull request merged**